### DISCUSSION TITLE
setCookie with supabase-js v2

### DISCUSSION THREAD

**Discussion by alexverdaguer** [supabase/supabase#8420] [General]
With V1 I used await supabase.auth.api.setAuthCookie(req, res) but with V2 I don't see this option using admin, like: await supabase.auth.admin.setAuthCookie(req, res);

Is there an option with V2 ? I would like to implement it with Nextjs and middleware.

Thank you

**Comment by hjaber** [comment]
Thanks for bringing this up, I had middleware setup with Nextjs that was using supabase.auth.api.setAuthCookie(req,res) and am trying to figure out the alternative to the breaking change.

**Comment by GaryAustin1** [comment]
This was the comment on the change (pull request) FWIW.

_Cookie-related methods and helpers should be implemented where they are used. For example, in the [auth-helpers](https://github.com/supabase/auth-helpers) library directly._

I guess looking into how auth-helpers deals with it would be a stop, might also raise an issue in supabase/gotrue-js seeking advice.

I don't use cookies, so can't help further.

**Reply by hjaber** [reply]
Thanks for the context @GaryAustin1, I'll take a look at the auth-helpers library and try to recreate a custom cookie.

While the [auth-helpers](https://github.com/supabase/auth-helpers) library is really helpful, it is opinionated and a black box that is difficult to debug.
